docs/snapcraft-parts.md

```diff
@@ -0,0 +1,126 @@
+# Snapcraft parts
+
+Parts are the main building block to create snaps using Snapcraft. Parts have
+their own private space and lifecycle. Each part uses a `plugin`, which tells
+the part how to behave and what to do with the information inside it.
+
+As seen in the [article about snapcraft.yaml syntax](snapcraft-syntax.md)
+parts have general keywords that apply to all of them. In one case, you may
+want to enhance your part's functionality using `stage-packages` which end up
+bringing Ubuntu deb-based packages into your part, `filesets` to declare
+inclusion and exclusion sets, `organize` so make the artifact output for your
+part neater, `stage` and `snap` to make certain only the right set of files is
+seen at each stage (making use of `filesets` or not). An example integrating
+these concepts for a part called `example-part` using a hypothetical plugin
+called `null` would look like:
+
+	parts:
+	  example-part:
+	    type: null
+	    stage-packages:
+	      - gpg
+	      - wget
+	    organize:
+	      opt/bin: bin
+	    filesets:
+	      binaries:
+	       - bin/*
+	       - usr/bin/*
+	      headers:
+	       - *.h
+	       - -include
+	    stage:
+	      - $binaries
+	      - test/bin/test_app
+	      - $headers
+	    snap:
+	      - $binaries
+ 
+In this example, imagine that the `null` plugin actually builds something in
+its private *build* location using its private *source* directory as a base,
+and that it installs the usual set of files to the file, except it will 
+*install* the part's private install directory.
+
+This `null` plugin makes use of `stage-packages`, these packages will be
+fetched from the Ubuntu deb archive using the *series* (release, i.e.; trusty,
+vivid, wily, ...) that is being used on the host. In this case, the part will
+be enhanced by the *gpg* and *wget* deb packages and its necessary
+dependencies to work isolated inside the part.
+
+When reaching the *stage* phase, the components in the private part's
+*install* directory will be exposed there, but since we used the organize
+keyword the contents in the install directory will be exposed to other parts
+in a cleaner form if desired or required; it is important to notice that in
+the event of using `filesets` they will follow the organized files and not
+the internal layout.
+
+The concept of `filesets` basically allows the creation of sets named after
+the keywords defined within, in this case *binaries* and *headers*, these are
+not necessarily needed but allow for variable expansion in the common
+targets: `stage` and `snap`. An inclusion is defined by just listing the
+target file, it can be globbed with `*` and a file can be explicitly
+excluded by prepending a `-` (when using `*` at the beginning of a path it
+needs to be quoted).
```

Comment:
I didn't understand this part. Can you please add an example of when it needs to be quoted?